src/test-helpers/typeFactory.ts

```diff
@@ -73,10 +74,14 @@ export class TypeFactory {
 	storageKey(
 		index: number,
 		indexType: keyof InterfaceTypes,
-		storageEntry: StorageEntryBase<'promise', GenericStorageEntryFunction>
+		storageEntry: StorageEntryBase<'promise', GenericStorageEntryFunction>,
+		secondaryId?: AccountId
```

Comment:
This needs to be changed to an `Options?` where it takes in any secondary id to be used in the StorageKey. The only issue is when I wrote this I wasnt sure how to access all polkadot-js types within one type check. The naive way, and way I might have to do is take in a `keyof interfaces` and perform the logic inside of the function itself to allocate a secondary key or not.

Reply:
You could also create a second function for double map storage keys

Reply:
Awesome, took your suggestion and made a `storageKeyDoubleMap` thanks!